lib/ansible/modules/network/onyx/onyx_username.py

```diff
@@ -0,0 +1,290 @@
+            - Decide the type of setted password (plain text or encrypted)
+        type: bool
+        default: False
+    capability:
+        description:
+            - Grant capability to this user account
+        type: str
+        choices: ['monitor', 'unpriv', 'v_admin', 'admin']
+    reset_capability:
+        description:
+            - Reset capability to this user account
+        type: bool
+        default: False
+    disconnected:
+        description:
+            - Disconnect all sessions of this user
+        type: bool
+        default: False
+    disabled:
+        description:
+            - Disable means of logging into this account
+        type: str
+        choices: ['none', 'login', 'password', 'all']
+    state:
+        description:
+            - Set state of the given account
+        default: present
+        type: str
+        choices: ['present', 'absent']
+"""
+
+EXAMPLES = """
+- name: create new user
+  onyx_username:
+      username: anass
+
+- name: set the user full-name
+  onyx_username:
+      username: anass
+      full_name: anasshami
+
+- name: set the user encrypted password
+  onyx_username:
+      username: anass
+      password: 12345
+      encrypted_password: True
+
+- name: set the user capability
+  onyx_username:
+      username: anass
+      capability: monitor
+
+- name: reset the user capability
+  onyx_username:
+      username: anass
+      reset_capability: True
+
+- name: remove the user configuration
+  onyx_username:
+      username: anass
+      state: absent
+"""
+
+RETURN = """
+commands:
+  description: The list of configuration mode commands to send to the device.
+  returned: always
+  type: list
+  sample:
+    - username *
+    - username * password *
+    - username * nopassword
+    - username * disable login
+    - username * capability admin
+    - no username *
+    - no username * disable
+
+"""
+import re
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.onyx.onyx import BaseOnyxModule, show_cmd
+
+
+class OnyxUsernameModule(BaseOnyxModule):
+    ACCOUNT_STATE = {
+        'Account locked out': dict(disabled='all'),
+        'No password required for login': dict(nopassword=True),
+        'Local password login disabled': dict(disabled='password'),
+        'Account disabled': dict(disabled='all')
+    }
+    ENCRYPTED_ID = 7
+
+    def init_module(self):
+        """
+        module initialization
+        """
+        element_spec = dict()
+
+        argument_spec = dict(state=dict(choices=['absent', 'present'], default='present'),
+                             username=dict(type='str', required=True),
+                             disabled=dict(choices=['none', 'login', 'password', 'all']),
+                             capability=dict(choices=['monitor', 'unpriv', 'v_admin', 'admin']),
+                             nopassword=dict(type='bool', default=False),
+                             password=dict(type='str'),
```

Comment:
Add `no_log=True` in argspec for password related options

Reply:
I'll add it 
Thanks,